### ISSUE TITLE
sequence lifeline height badly depends on message height

### ISSUE THREAD

**Issue by jhoehle** [plantuml/plantuml#1099]
When connecting arrow messages with &, the height of activation lines is miscomputed except in the simple case where all message texts share the same text height.
```
@startuml
!pragma teoz true
  A <[#red]-o? ++
  A -> B -- : very long\nmulti-line
& B -> C ++ #red: single
note over C: got it
Deactivate C
====
  A <-o? ++ #green
  A -> B -- : very long\nmulti-line
& B -> C ++ #green : same\nheight
note over C: got it
Deactivate C
====
  A <-o? ++ #red
  A -> B -- : single
& B -> C ++ : very long\nmulti-line
note over C: got it
Deactivate C
@enduml
```
![sequence diagram](https://www.plantuml.com/plantuml/png/dS_1QiCm30RWkv_Y7u5TWbyWR5csx2sM7SmY78Cj3qSEj4zVfpxbK4ghewIVJnfMCKboyEhbFnaR38JY6P8oAU0JRpzDelbNnm-qxRMZUtn1QtJOA9tW8zk9G_RYj7TCwdKVznZBFlPiXzMnzQGu2Y6M4COEDWgSg6yoV-8s8uHHlPUgQENGs4J43wirXWARG1ClvEmYp_29vYDUNpbeDo-wwmt4Smx-0W00)

The third case demonstrates how the height of a lifeline must depend on all the following &-connected arrows.
Work-around: Manually adjust all texts to the same height, e.g. using `\nsingle`. Using that trick, I could correct _some_ misplaced lifelines in my diagrams.

Bug too: With Teoz, `<-o?` without text looks awfully short. The standard engine produces a nice view.

Documentation bug: [The online documentation](https://plantuml.com/sequence-diagram) uses &-notation in section "Slanted or odd arrows" without introducing it. Combining arrows using & should be added to documentation bug #261

**Comment by jubr** [comment]
Hi @The-Lum, would it be possible to take a look at this one please? This teoz bug been around for ages and I'm still running into this on a daily basis :'(
Please let me know if more illustrative examples are required, tnx!

**Comment by jimnelson372** [comment]
I have submitted the fix in pull request #1793.   With this fix, your example generates:

![testteozlifeline_005](https://github.com/plantuml/plantuml/assets/66284321/8e607076-d472-4ed6-adfd-30d15558b8dd)

Regards,

Jim Nelson
